bc_obps/reporting/service/compliance_service.py

```diff
@@ -186,17 +188,26 @@ def calculate_product_emission_limit(
         )
         return product_emission_limit
 
-    @classmethod
-    def get_calculated_compliance_data(cls, report_version_id: int) -> ComplianceData:
+    @staticmethod
+    def get_calculated_compliance_data(report_version_id: int) -> ComplianceData:
+        # Fetch the ReportVersion once (bring in reporting_year and operation) to avoid extra queries
+        report_version_record = ReportVersion.objects.select_related("report__reporting_year", "report__operation").get(
+            pk=report_version_id
+        )
+
         naics_data = ComplianceService.get_regulatory_values_by_naics_code(report_version_id)
-        registration_purpose = ReportVersion.objects.get(pk=report_version_id).report.operation.registration_purpose
+        registration_purpose = report_version_record.report.operation.registration_purpose
```

Comment:
This should be fetched from the report_operation record from that report version (we can add it to the `select_related` above too)
```suggestion
        registration_purpose = report_version_record.report_operation.registration_purpose
```